Unreal/CarlaUE4/Plugins/CarlaTools/Source/CarlaTools/Private/OpenDriveToMap.cpp

```diff
@@ -154,16 +154,59 @@ void UOpenDriveToMap::CreateMap()
     UE_LOG(LogCarlaToolsMapGenerator, Error, TEXT("Map Name Is Empty") );
     return;
   }
-  if ( !IsValid(FileDownloader) )
-  {
-    FileDownloader = NewObject<UCustomFileDownloader>();
-  }
 
-  FileDownloader->ResultFileName = MapName;
-  FileDownloader->Url = Url;
+  if( !Url.IsEmpty() ) {
+    if ( !IsValid(FileDownloader) )
+    {
+      FileDownloader = NewObject<UCustomFileDownloader>();
+    }
 
-  FileDownloader->DownloadDelegate.BindUObject( this, &UOpenDriveToMap::ConvertOSMInOpenDrive );
-  FileDownloader->StartDownload();
+    FileDownloader->ResultFileName = MapName;
+    FileDownloader->Url = Url;
+
+    FileDownloader->DownloadDelegate.BindUObject( this, &UOpenDriveToMap::ConvertOSMInOpenDrive );
+    FileDownloader->StartDownload();
+  }else{
+    if(LocalFilePath.Contains(".xodr"))
+    {
+      IPlatformFile& FileManager = FPlatformFileManager::Get().GetPlatformFile();
```

Comment:
### Clean Code
To split this big function I suggest the next structure:

```c++
else if(LocalFilePath.EndsWith(".xodr"))
{
  importXodr(LocalFilePath);
} else if(LocalFilePath.EndsWith(".osm"))
  importOsm(LocalFilePath);
}
```

Reply:
Done